play-media-native/screens/AddSport.tsx

```diff
@@ -0,0 +1,156 @@
+import { Screen } from '../features/Screen/Screen';
+import { useContentItems } from '../hooks/useContentItems/useContentItems';
+import { useScrollOffset } from '../hooks/useScrollOffset/useScrollOffset';
+import { useSportsQuery } from '../hooks/useSportsQuery/useSportsQuery';
+import { Sport } from '../interfaces/sport';
+import { styles } from '../theme/styles';
+import { theme } from '../theme/theme';
+
+export const AddSportsScreen = ({ navigation, route }) => {
+  const fieldKey = route?.params?.key;
+  const initialRoute = route?.params?.initialRoute;
+  const stateKey = route?.params?.stateKey;
+  const single = route?.params?.single;
+
+  const { contentItems, edit: editFields } = useContentItems();
+
+  const {
+    data: sports,
+    isLoading: isFetchingInitialSports,
+    refetch: refetchSports,
+    isRefetching: isRefetchingSports,
+  } = useSportsQuery();
+  const { calcScrollOffset } = useScrollOffset(true);
+  const initialSelectedIDs = useMemo(() => {
+    if (Array.isArray(contentItems[stateKey][fieldKey])) {
+      return contentItems[stateKey][fieldKey].map((item: Sport) => item.id);
+    }
+
+    return contentItems[stateKey][fieldKey]?.id || [];
+  }, [contentItems, fieldKey, stateKey]);
+  const [selectedSportIDs, setSelectedSportIDs] = useState<string[]>([initialSelectedIDs]);
+  const noneSelected = !selectedSportIDs?.length;
+
+  const edit = useCallback(
+    ({ key, value }: { key: string; value: Sport[] }) => {
+      editFields({ id: stateKey, key, value: [...contentItems[stateKey][key], ...value] });
+    },
+    [contentItems, editFields, stateKey]
+  );
+
+  const handleRefresh = useCallback(() => {
+    refetchSports();
+  }, [refetchSports]);
+
+  const onSelect = useCallback((sport: Sport) => {
+    setSelectedSportIDs((prevSelectedSportIDs) => {
+      if (prevSelectedSportIDs.includes(sport.id)) {
+        return prevSelectedSportIDs.filter((item) => item !== sport.id);
+      }
+
+      return [...prevSelectedSportIDs, sport.id];
+    });
+  }, []);
+
+  const onSelectSingle = useCallback(
+    (sport: Sport) => {
+      editFields({
+        id: stateKey,
+        key: route.params.key,
+        value: sport,
+      });
+
+      navigation.navigate(initialRoute);
+    },
+    [editFields, initialRoute, navigation, route.params.key, stateKey]
+  );
+
+  const onCancel = useCallback(() => {
+    navigation.goBack();
+  }, [navigation]);
+
+  const onSubmit = useCallback(() => {
+    if (!fieldKey || !initialRoute) {
+      return;
+    }
+
+    edit({
+      key: fieldKey,
+      value: sports.filter((item) => selectedSportIDs.includes(item.id)),
+    });
+
+    navigation.navigate(initialRoute);
+  }, [edit, fieldKey, initialRoute, navigation, selectedSportIDs, sports]);
+
+  if (single) {
+    return (
+      <Screen>
+        <Listing
+          data={sports}
+          isLoading={isFetchingInitialSports}
+          numColumns={2}
+          renderItem={({ item }) => (
+            <CardSport
```

Comment:
There is something wrong with the Listing. Depending on what I have selected different sports appear. 
[Nothing selected](https://scontent.fsof8-1.fna.fbcdn.net/v/t1.15752-9/334926423_197827312856969_2148915917303426376_n.jpg?_nc_cat=103&ccb=1-7&_nc_sid=ae9488&_nc_ohc=193dvMNuj6oAX_IJxA1&_nc_ht=scontent.fsof8-1.fna&oh=03_AdRPWPUECeYt8DKWsPyAOHJBL2b-EyZfYntNEZaO5uf8rA&oe=642D23D7)
[Skateboarding selected](https://scontent.fsof8-1.fna.fbcdn.net/v/t1.15752-9/334927897_442254234733708_157601388328995673_n.jpg?_nc_cat=101&ccb=1-7&_nc_sid=ae9488&_nc_ohc=NvSkqMciR1kAX9PfKYD&_nc_ht=scontent.fsof8-1.fna&oh=03_AdS_DGBNZ5WGrdY465tcQhUsyJ3X9WQ_icBrPbab7gWzGA&oe=642CF43C)
[Climbing selected](https://scontent.fsof8-1.fna.fbcdn.net/v/t1.15752-9/334880890_737352741131892_7410477830176955600_n.jpg?_nc_cat=102&ccb=1-7&_nc_sid=ae9488&_nc_ohc=Q3fciMf4QQsAX-PxK1H&_nc_ht=scontent.fsof8-1.fna&oh=03_AdQZsHRrcVCMdFodBmZSQkcDQSE6LpllcHrm9k8ClqvN1w&oe=642D0638)

When dragging down to refresh the listing gets cut off (not present in event and athlete listing screens).
[Refreshing](https://scontent.fsof8-1.fna.fbcdn.net/v/t1.15752-9/334901960_560136875902000_28462774421278257_n.jpg?_nc_cat=102&ccb=1-7&_nc_sid=ae9488&_nc_ohc=BOsP9Lq4UIMAX_sv4Ax&_nc_ht=scontent.fsof8-1.fna&oh=03_AdTufE5dCGhgJIBHmqN6gh6R0tDKeXYMW1tPixAuyeAkbA&oe=642D1DDE)

Also the numColumns don't seem to be working properly.

Reply:
I pushed a fix that could possibly fix things on iOS, but I can't reproduce these on Android. This is what I see.

Selected sport
![sport-selected](https://user-images.githubusercontent.com/110840437/223087253-0adb21df-2dff-4af9-830c-714e6a218607.jpg)

None selected
![none-selected](https://user-images.githubusercontent.com/110840437/223087310-8c080db0-876a-4e71-8678-81e1a3e5e645.jpg)

Refreshing
![refreshing](https://user-images.githubusercontent.com/110840437/223087365-ca4181b9-b32d-49f8-a62b-0af80802ccf5.jpg)